d2ir/compile.go

```diff
@@ -145,20 +152,34 @@ func (c *compiler) resolveSubstitutions(varsStack []*Map, node Node) {
 				}
 				if box.Substitution.Spread {
 					if resolvedField.Composite == nil {
-						c.errorf(box.Substitution, "cannot spread non-map into map")
+						c.errorf(box.Substitution, "cannot spread non-composite into composite")
```

Comment:
"cannot spread non-composite."

Reply:
Like you can't spread a non composite period, not only into a composite.

Reply:
ah right